common/djangoapps/monkey_patch/django_db_models_options.py

```diff
@@ -0,0 +1,45 @@
+"""
+Monkey patch implementation of the following _expire_cache performance improvement:
+
+https://github.com/django/django/commit/7628f87e2b1ab4b8a881f06c8973be4c368aaa3d
+
+Remove once we upgrade to a version of django which includes this fix natively!
+"""
+
+from django.db.models.options import Options
+
+
+def patch():
+    """
+    Monkey-patch the Options class.
+    """
+    def _expire_cache(self, forward=True, reverse=True):
+        # This method is usually called by apps.cache_clear(), when the
+        # registry is finalized, or when a new field is added.
+        if forward:
+            for cache_key in self.FORWARD_PROPERTIES:
+                if cache_key in self.__dict__:
+                    delattr(self, cache_key)
+        if reverse and not self.abstract:
+            for cache_key in self.REVERSE_PROPERTIES:
+                if cache_key in self.__dict__:
+                    delattr(self, cache_key)
+        self._get_fields_cache = {}
+
+    # Patch constants with dict object instead of a set.
+    Options.FORWARD_PROPERTIES = {
```

Comment:
The curly braces are Python set literal syntax. For more, see http://www.renzolucioni.com/pythons-set-literals/.